### PR TITLE
Remove deprecated Ruby File.exists? in helper script

### LINKED ISSUE #109385: `podhelper.rb` uses deprecated, recently-removed `exists?` method
Automatically signing iOS for device deployment using specified development team
in Xcode project: 89B89GC38P
Running pod install...                                             324ms
CocoaPods' output:
↳

    [!] Invalid `Podfile` file: undefined method `exists?' for File:Class

      return [] unless File.exists? file_path
                           ^^^^^^^^
    Did you mean?  exist?.

     #  from /Users/project/new1/ios/Podfile:38
     #  -------------------------------------------
     #    use_modular_headers!
     >    flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
     #  end
     #  -------------------------------------------

### PR DESCRIPTION
**Pull request by jmagman**
`File.exists?` has evidently been deprecated in Ruby 2.1 and was removed in Ruby 3.2 preview 1 (3.2 has not yet been released). https://bugs.ruby-lang.org/issues/17391
https://www.ruby-lang.org/en/news/2022/04/03/ruby-3-2-0-preview1-released/

I ran the [RuboCop](https://docs.rubocop.org/) linter on the two Ruby files in our repo, which caught the deprecation, as well as some outdated syntax.

On `podhelper.rb`:
```
podhelper.rb:17:1: C: [Corrected] Style/EmptyMethod: Put empty method definitions on a single line.
def flutter_ios_podfile_setup ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:21:1: C: [Corrected] Style/EmptyMethod: Put empty method definitions on a single line.
def flutter_macos_podfile_setup ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:63:44: C: [Corrected] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      next if xcframework_file.start_with?(".") # Hidden file, possibly on external disk.
                                           ^^^
podhelper.rb:64:37: C: [Corrected] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      if xcframework_file.end_with?("-simulator") # ios-arm64_x86_64-simulator
                                    ^^^^^^^^^^^^
podhelper.rb:66:42: C: [Corrected] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      elsif xcframework_file.start_with?("ios-") # ios-arm64
                                         ^^^^^^
podhelper.rb:68:7: C: [Corrected] Style/EmptyElse: Redundant else-clause.
      else
      ^^^^
podhelper.rb:161:70: C: [Corrected] Style/RedundantSelf: Redundant self detected.
  ios_application_path ||= File.dirname(defined_in_file.realpath) if self.respond_to?(:defined_in_file)
                                                                     ^^^^
podhelper.rb:170:39: C: [Corrected] Style/BlockDelimiters: Avoid using {...} for multi-line blocks.
  File.open(copied_podspec_path, 'w') { |podspec|
podhelper.rb:194:18: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  pod 'Flutter', :path => flutter_relative_path_from_podfile(podspec_directory)
                 ^^^^^^^^
podhelper.rb:200:70: C: [Corrected] Style/RedundantSelf: Redundant self detected.
  mac_application_path ||= File.dirname(defined_in_file.realpath) if self.respond_to?(:defined_in_file)
                                                                     ^^^^
podhelper.rb:208:39: C: [Corrected] Style/BlockDelimiters: Avoid using {...} for multi-line blocks.
  File.open(copied_podspec_path, 'w') { |podspec|
                                      ^
podhelper.rb:232:23: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  pod 'FlutterMacOS', :path => File.join('Flutter', 'ephemeral')
podhelper.rb:241:66: C: [Corrected] Style/RedundantSelf: Redundant self detected.
  application_path ||= File.dirname(defined_in_file.realpath) if self.respond_to?(:defined_in_file)
                                                                 ^^^^
podhelper.rb:255:17: C: [Corrected] Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an unless.
    next unless (plugin_name && plugin_path && has_native_build)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:259:5: C: [Corrected] Style/Next: Use next to skip iteration.
    if (plugin_name && plugin_path && has_native_build)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:259:8: C: [Corrected] Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an if.
    if (plugin_name && plugin_path && has_native_build)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:266:24: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
      pod plugin_name, :path => File.join(relative, platform)
                       ^^^^^^^^
podhelper.rb:275:25: W: [Corrected] Lint/DeprecatedClassMethods: File.exists? is deprecated in favor of File.exist?.
  return [] unless File.exists? file_path
                        ^^^^^^^
podhelper.rb:276:38: C: [Correctable] Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
  return [] unless dependencies_hash.has_key?('plugins')
                                     ^^^^^^^^
podhelper.rb:277:49: C: [Correctable] Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
  return [] unless dependencies_hash['plugins'].has_key?('ios')
```
and the template script:
```
podhelper.rb:70:34: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  pod 'FlutterPluginRegistrant', :path => File.join(relative, 'Flutter', 'FlutterPluginRegistrant'), :inhibit_warnings => true
                                 ^^^^^^^^
podhelper.rb:70:102: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  pod 'FlutterPluginRegistrant', :path => File.join(relative, 'Flutter', 'FlutterPluginRegistrant'), :inhibit_warnings => true
                                                                                                     ^^^^^^^^^^^^^^^^^^^^
podhelper.rb:90:103: C: [Corrected] Style/Semicolon: Do not use semicolons to terminate expressions.
  flutter_export_environment_path = File.join('${SRCROOT}', relative, 'flutter_export_environment.sh');
                                                                                                      ^
podhelper.rb:93:16: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  script_phase :name => 'Run Flutter Build {{projectName}} Script',
               ^^^^^^^^
podhelper.rb:94:5: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
    :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh build",
    ^^^^^^^^^^
podhelper.rb:95:5: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
    :execution_position => :before_compile
    ^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:98:16: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
  script_phase :name => 'Embed Flutter Build {{projectName}} Script',
               ^^^^^^^^
podhelper.rb:99:5: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
    :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh embed_and_thin",
    ^^^^^^^^^^
podhelper.rb:100:5: C: [Corrected] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
    :execution_position => :after_compile
    ^^^^^^^^^^^^^^^^^^^^^^
podhelper.rb:105:3: C: [Corrected] Style/IfUnlessModifier: Favor modifier unless usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
  unless File.exist?(generated_xcode_build_settings_path)
  ^^^^^^
podhelper.rb:110:39: C: [Corrected] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
                                      ^^
podhelper.rb:130:42: W: [Corrected] Lint/UnusedBlockArgument: Unused block argument - build_configuration. You can omit the argument if you don't care about it.
    target.build_configurations.each do |build_configuration|
                                         ^^^^^^^^^^^^^^^^^^^
```

We have good test coverage of both these methods, but they are running on a stable version of Ruby in our CI (I think it's just using the macOS default system version 2.6).   
Getting a preview version of Ruby into our CI system doesn't seem feasible, so filed https://github.com/flutter/flutter/issues/109431 to get a linter running in CI to catch deprecations like this.

Fixes https://github.com/flutter/flutter/issues/109385

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/bin/podhelper.rb [hunk 1]
```diff
@@ -256,23 +253,22 @@ def flutter_install_plugin_pods(application_path = nil, relative_symlink_dir, pl
...
+    pod plugin_name, path: File.join(relative, platform)
   end
 end
 
 # .flutter-plugins-dependencies format documented at
 # https://flutter.dev/go/plugins-list-migration
 def flutter_parse_plugins_file(file, platform)
   file_path = File.expand_path(file)
-  return [] unless File.exists? file_path
+  return [] unless File.exist? file_path
```
Comment (jmagman) [inline packages/flutter_tools/bin/podhelper.rb:271]: This was the actual fix, `exists?` is deprecated.

**Review by christopherfujino — APPROVED** [review]
i don't think `.exist?` is better than `.exists?`, but nonetheless LGTM

**Comment by jmagman** [comment]
> i don't think `.exist?` is better than `.exists?`, but nonetheless LGTM

`.exists?` is deprecated, that's the bug this is fixing https://github.com/flutter/flutter/issues/109385. Take it up with the ruby lang folks I guess 🤷‍♀️

**Comment by christopherfujino** [comment]
> > i don't think `.exist?` is better than `.exists?`, but nonetheless LGTM
> 
> `.exists?` is deprecated, that's the bug this is fixing #109385. Take it up with the ruby lang folks I guess woman_shrugging

don't mind me...
oldmanyellsatcloud

**Comment by Hixie** [comment]
test-exempt: for practical reasons. testing this is likely to cost more than it saves since it's an entirely new ecosystem. https://github.com/flutter/flutter/issues/109431 in case we change out minds.

**Comment by jmagman** [comment]
Note to future self:

flutter/packages/flutter_tools/bin/.rubocop.yml
```yml
AllCops:
  TargetRubyVersion: 2.5
Layout:
  Enabled: false
```

**Pull request merged**